scrapy_poet/injection.py

```diff
@@ -142,14 +162,44 @@ def build_instances_from_providers(
             if not provided_classes:
                 continue
 
-            kwargs = andi.plan(
-                provider,
-                is_injectable=is_injectable,
-                externally_provided=scrapy_provided_dependencies,
-                full_final_kwargs=False,
-            ).final_kwargs(scrapy_provided_dependencies)
-            objs = yield maybeDeferred_coro(provider, set(provided_classes),
-                                               **kwargs)
+            objs, fingerprint = None, None
+            cache_hit = False
+            if self.cache and provider.has_cache_support:
+                if not provider.name:
+                    raise NotImplementedError(f"The provider {type(provider)} must have a `name` defined if"
+                                              f" you want to use the cache. It must be unique across the providers.")
+                # Return the data if it is already in the cache
+                fingerprint = f"{provider.name}_{provider.fingerprint(set(provided_classes), request)}"
```

Comment:
(very minor) What do you think about extracting a method which computes these fingerprints? Use case: get a cache key programmatically, if you know the provider.

Reply:
Also, an idea, probably for the future: store cache for different providers in different tables (sqlitedict supports this via "tablename" argument, when SqliteDict is created). The main advantage would be that fingerprints would remain json. sqlite may even be able to query inside, via json1 extension. But that's all not needed now :)

Reply:
Got it. Makes sense. I've created a draft PR here before these ideas get buried down. 😄 https://github.com/scrapinghub/scrapy-poet/pull/59